compliance/collection/nodeinventorizer/caching_inventorizer.go

```diff
@@ -99,15 +99,18 @@ func min(d1 time.Duration, d2 time.Duration) time.Duration {
 	return d1
 }
 
-func readBackoff(path string) time.Duration {
+// readBackoff reads a backoff from a location. It will return either the loaded time.Duration, 0 if it doesn't exist, or errorBackoff on errors.
+func readBackoff(path string, errorBackoff time.Duration) time.Duration {
 	wrap := readInventoryWrap(path)
 	if wrap == nil {
 		return 0
 	}
```

Comment:
This branch remains unaddressed. There may be two cases when `readInventoryWrap()` returns `nil`: the file is not there or there was an error reading or parsing file. It is valid to return `0` in case of absent file, but `0` is wrong in case of error. That's what I meant by differentiating in my previous comment https://github.com/stackrox/stackrox/pull/4701#discussion_r1131263147

Reply:
I see, thank you. I have added error handling in [e555489](https://github.com/stackrox/stackrox/pull/4701/commits/e5554895838b60dcded83028d614ffb46c2a3445).
If we have an error reading the backoff, it returns `maxBackoff`.
If we have an error reading the cached inventory, we write a backoff file with `maxBackoff` and continue with collecting a new inventory after waiting.

Reply:
Looks good after the change.